app/services/katello/repository_type_manager.rb

```diff
@@ -31,14 +31,16 @@ def fix_pulp3_capabilities
       end
 
       def enabled_repository_types(update = true)
-        if update && pulp_primary&.has_feature?(PULP3_FEATURE) && pulp_primary&.capabilities(PULP3_FEATURE)&.empty?
-          fix_pulp3_capabilities
-        end
+        return @enabled_repository_types unless update
```

Comment:
This flag wasn't doing what it was designed to do..With the update flag defaulting to true, setting it to flase should mean the enabled_repository_types are returned as is.